### PR TITLE
[MRG] Fix issue 2108 dipole

### LINKED ISSUE #2108: indexing Dipole
it would be great to be able index instances of Dipole e.g. to do:

`dipole[dipole.gof > 0.8].plot_locations()`

we should be able to do:

```
dipole[dipole.gof > 0.8]
dipole[:5]
```

as if we had 1d numpy array of dipoles.

@jmontoyam It would simplify your scripts.

### PR DESCRIPTION
**Pull request by agramfort**
fixes #2108 

taking over #2147

cc @jmontoyam

### CODE REVIEW

**Comment by agramfort** [comment]
ready to go on my side

File: mne/dipole.py [hunk 1]
```diff
@@ -190,6 +190,27 @@ def plot_amplitudes(self, color='k', show=True):
...
+    def __getitem__(self, idx_slice):
+        """Handle indexing"""
+        if isinstance(idx_slice, int):  # make sure attributes stay 2d
+            idx_slice = [idx_slice]
+
+        selected_times = self.times[idx_slice]
+        selected_pos = self.pos[idx_slice, :]
+        selected_amplitude = self.amplitude[idx_slice]
+        selected_ori = self.ori[idx_slice, :]
+        selected_gof = self.gof[idx_slice]
```
Comment (Eric89GXL) [inline mne/dipole.py]: these should all probably be `.copy()`'ed

**Comment by Eric89GXL** [comment]
Other than my minor gripe about copies of the attributes LGTM

**Comment by agramfort** [comment]
done

**Comment by agramfort** [comment]
looks like we're good here...

**Comment by Eric89GXL** [comment]
Indeed, thanks @agramfort

**Pull request merged**